### ISSUE TITLE
Fix inaccessible uppercase text

### ISSUE THREAD

**Issue by felixarntz** [GoogleForCreators/web-stories-wp#6015]
## Bug Description

Web Stories currently relies on fully capitalized DOM content in order to display certain text as uppercase. This is bad for accessibility, since screen readers will read a text like "FOOD" out as if it was an abbreviation ("F-O-O-D"). Such a visual appearance should instead be achived by using regularly capitalized text ("Food") and use CSS `text-transform:uppercase`.

Addressing this will probably include two steps at a high level:
* Introducing a UI text control that allows showing text all uppercase. This would then need to result in a `text-transform:uppercase` CSS rule being applied to the element.
* Correcting all currently fully capitalized text in Web Stories templates shipped so that they instead rely on the new appearance option.

Unfortunately this will not fix existing stories that already have added fully uppercase text to work around this limitation. Since the negative accessibility impact of this on stories on the web will therefore only grow with time, it may be worth prioritizing this work.

## Expected Behaviour

Uppercase text should be achieved via CSS instead of actually having the text all uppercase.

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Implementation Brief
Add a new [`formatter`](https://github.com/google/web-stories-wp/tree/main/assets/src/edit-story/components/richText/formatters) to the `customStyleFn` for the draft-js `Editor`.

Ensure the uppercase is applied inline and can be applied to part of the text only.

**Comment by felixarntz** [comment]
cc @swissspidy

**Comment by swissspidy** [comment]
> Unfortunately this will not fix existing stories that already have added fully uppercase text to work around this limitation. Since the negative accessibility impact of this on stories on the web will therefore only grow with time, it may be worth prioritizing this work.

We can add a migration for existing stories if desired.

**Comment by swissspidy** [comment]
cc @samitron7 for considerring adding a new button for uppercase text

**Comment by felixarntz** [comment]
@swissspidy 
> We can add a migration for existing stories if desired.

I thought about that too, but don't think it's really feasible because it's impossible to know (except with some advanced language interpreter engine) whether a word is _actually_ an abbreviation and should be capitalized or not 🤔

**Comment by swissspidy** [comment]
> @swissspidy
> 
> > We can add a migration for existing stories if desired.
> 
> I thought about that too, but don't think it's really feasible because it's impossible to know (except with some advanced language interpreter engine) whether a word is _actually_ an abbreviation and should be capitalized or not 🤔

True. Plus it doesn't stop users from WRITING LIKE THIS because they don't know about this button.

**Comment by swissspidy** [comment]
@aaecheverria I'd love to address this one. What we need is a new button in the design panel for making all text (or the selected text) uppercase

**Comment by aaecheverria** [comment]
@swissspidy 
Hi Pascal, just to make sure I'm understanding correctly. You mean a new button within the Text section in the design panel (please see image below)? 
<img width="354" alt="Captura de Pantalla 2021-05-19 a la(s) 9 45 59 a  m" src="https://user-images.githubusercontent.com/73601374/118833946-7b14f980-b887-11eb-813c-5c2645322ba5.png">

**Comment by ivayloklissarov** [comment]
I have updated the design system with this new element @swissspidy 

![image](https://user-images.githubusercontent.com/77846281/118843970-2cb82880-b890-11eb-83ef-d9b42894f672.png)

**Comment by swissspidy** [comment]
@aaecheverria Correct! @ivayloklissarov Yeah that's what I had in mind 👍 Is there enough space in that row to accommodate for that button? :)

**Comment by ivayloklissarov** [comment]
Yes I have put it on the far right.  If we want to include more options than what is currently shown on the row, we need to re-think the layout display. @swissspidy

**Comment by swissspidy** [comment]
Sounds good.

@miina How easy would it be to implement this? :-)

**Comment by miina** [comment]
@swissspidy Sounds like a `3` -- we'd need to add a new `formatter` to the `customStyleFn` for the draft-js `Editor`, considering selecting only part of the text and applying uppercase needs to be possible.

**Comment by swissspidy** [comment]
Sounds good 👍 

Removing the UX Needed label so you could move this on the board wherever.

Opened #7658 to update the templates once we have this new functionality.

**Comment by csossi** [comment]
Verified in QA
